fc/mef.py

```diff
@@ -736,230 +506,223 @@ def get_transform_fxn(data_beads, peaks_mef, mef_channels,
     if plot_filename is None:
         plot_filename = str(data_beads)
 
-    # 1. Cluster
-    # ===========
-    if cluster_method == 'dbscan':
-        labels = clustering_dbscan(data_beads[:,cluster_channels], 
-            **cluster_params)
-    elif cluster_method == 'distance':
-        labels = clustering_distance(data_beads[:,cluster_channels], 
-            **cluster_params)
-    elif cluster_method == 'gmm':
-        labels = clustering_gmm(data_beads[:,cluster_channels], 
-            **cluster_params)
+    # mef_channels and mef_values should be iterables.
+    if hasattr(mef_channels, '__iter__'):
+        mef_channel_all = list(mef_channels)
+        mef_values_all = np.array(mef_values).copy()
     else:
-        raise ValueError("Clustering method {} not recognized."
-            .format(cluster_method))
+        mef_channel_all = [mef_channels]
+        mef_values_all = np.array([mef_values])
+
+    ###
+    # 1. Clustering
+    ###
 
+    # If clustering channels not specified, use channels in mef_channels
+    if clustering_channels is None:
+        clustering_channels = mef_channels
+
+    # Get number of clusters from number of specified MEF values
+    n_clusters = mef_values_all.shape[1]
+
+    # Run clustering function
+    labels = clustering_func(data_beads[:, clustering_channels],
+                             n_clusters,
+                             **clustering_params)
+
+    # Separate events corresponding to each cluster
     labels_all = np.array(list(set(labels)))
```

Comment:
is `labels_all` just trying to get all the unique labels? Might consider `unique_labels` instead of `labels_all` for clarity and `np.unique()` instead of invoking `set()`, `list()`, and `np.array()`.